A lot of java.lang.IllegalStateExceptions since 1.0.14

**java.lang.RuntimeException:** 

> at android.app.ActivityThread.handleReceiver (ActivityThread.java:3230)
at android.app.ActivityThread.-wrap17 (Unknown Source)
at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1698)
at android.os.Handler.dispatchMessage (Handler.java:106)
at android.os.Looper.loop (Looper.java:176)
at android.app.ActivityThread.main (ActivityThread.java:6635)
at java.lang.reflect.Method.invoke (Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:547)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:823)

**Caused by: java.lang.IllegalStateException:** 

> at android.app.ContextImpl.startServiceCommon (ContextImpl.java:1531)
at android.app.ContextImpl.startService (ContextImpl.java:1487)
at android.content.ContextWrapper.startService (ContextWrapper.java:660)
at android.content.ContextWrapper.startService (ContextWrapper.java:660)
at com.android.calendar.alerts.AlertReceiver.beginStartingService (AlertReceiver.java:137)
at com.android.calendar.alerts.AlertReceiver.onReceive (AlertReceiver.java:840)
at android.app.ActivityThread.handleReceiver (ActivityThread.java:3223)

--- a/src/com/android/calendar/alerts/AlertReceiver.java
+++ b/src/com/android/calendar/alerts/AlertReceiver.java
@@ -128,7 +128,7 @@ public class AlertReceiver extends BroadcastReceiver {
                 PowerManager pm =
                     (PowerManager)context.getSystemService(Context.POWER_SERVICE);
                 mStartingService = pm.newWakeLock(PowerManager.PARTIAL_WAKE_LOCK,
-                        "StartingAlertService");
+                        "Etar:StartingAlertService");
                 mStartingService.setReferenceCounted(false);
             }
             mStartingService.acquire();

--- a/src/com/android/calendar/alerts/AlertReceiver.java
+++ b/src/com/android/calendar/alerts/AlertReceiver.java
@@ -132,7 +132,13 @@ public class AlertReceiver extends BroadcastReceiver {
                 mStartingService.setReferenceCounted(false);
             }
             mStartingService.acquire();
-            context.startService(intent);
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                context.startForegroundService(intent);
+                //Log.d(TAG, "startForeground.");
+            } else {
+                context.startService(intent);
+            }
+
         }
     }
 

--- a/src/com/android/calendar/alerts/AlertReceiver.java
+++ b/src/com/android/calendar/alerts/AlertReceiver.java
@@ -132,9 +132,8 @@ public class AlertReceiver extends BroadcastReceiver {
                 mStartingService.setReferenceCounted(false);
             }
             mStartingService.acquire();
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            if (Utils.isOreoOrLater()) {
                 context.startForegroundService(intent);
-                //Log.d(TAG, "startForeground.");
             } else {
                 context.startService(intent);
             }

--- a/src/com/android/calendar/alerts/AlertService.java
+++ b/src/com/android/calendar/alerts/AlertService.java
@@ -982,14 +982,13 @@ public class AlertService extends Service {
     public int onStartCommand(Intent intent, int flags, int startId) {
         if (intent != null) {
 
-            if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
+            if (Utils.isOreoOrLater()) {
                 Notification notification = new NotificationCompat.Builder(this, ALERT_CHANNEL_ID)
                         .setContentTitle("Update alerts")
-                        .setSmallIcon(R.drawable.ic_alarm_white)
+                        .setSmallIcon(R.drawable.stat_notify_calendar)
                         .setShowWhen(false)
                         .build();
                 startForeground(1337, notification);
-                //Log.d(TAG, "startForeground did happen.");
             }
 
             Message msg = mServiceHandler.obtainMessage();

--- a/src/com/android/calendar/alerts/AlertService.java
+++ b/src/com/android/calendar/alerts/AlertService.java
@@ -984,7 +984,7 @@ public class AlertService extends Service {
 
             if (Utils.isOreoOrLater()) {
                 Notification notification = new NotificationCompat.Builder(this, ALERT_CHANNEL_ID)
-                        .setContentTitle("Update alerts")
+                        .setContentTitle("Event notifications")
                         .setSmallIcon(R.drawable.stat_notify_calendar)
                         .setShowWhen(false)
                         .build();

--- a/src/com/android/calendar/alerts/AlertService.java
+++ b/src/com/android/calendar/alerts/AlertService.java
@@ -173,7 +173,7 @@ public class AlertService extends Service {
         }
 
 
-        if (!prefs.getBoolean(GeneralPreferences.KEY_ALERTS, true)) {
+        if (!prefs.getBoolean(GeneralPreferences.KEY_ALERTS, true) && !Utils.isOreoOrLater()) {
             if (DEBUG) {
                 Log.d(TAG, "alert preference is OFF");
             }

--- a/src/com/android/calendar/alerts/AlertService.java
+++ b/src/com/android/calendar/alerts/AlertService.java
@@ -42,6 +42,7 @@ import android.os.Process;
 import android.provider.CalendarContract;
 import android.provider.CalendarContract.Attendees;
 import android.provider.CalendarContract.CalendarAlerts;
+import android.support.v4.app.NotificationCompat;
 import android.support.v4.content.ContextCompat;
 import android.text.TextUtils;
 import android.text.format.DateUtils;
@@ -980,6 +981,17 @@ public class AlertService extends Service {
     @Override
     public int onStartCommand(Intent intent, int flags, int startId) {
         if (intent != null) {
+
+            if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
+                Notification notification = new NotificationCompat.Builder(this, ALERT_CHANNEL_ID)
+                        .setContentTitle("Update alerts")
+                        .setSmallIcon(R.drawable.ic_alarm_white)
+                        .setShowWhen(false)
+                        .build();
+                startForeground(1337, notification);
+                //Log.d(TAG, "startForeground did happen.");
+            }
+
             Message msg = mServiceHandler.obtainMessage();
             msg.arg1 = startId;
             msg.obj = intent.getExtras();

